src/nodes.coffee

```diff
@@ -1416,6 +1423,19 @@ exports.Call = class Call extends Base
       fragments.push @makeCode(' />')
     fragments
 
+  astType: ->
+    if @isNew
+      'NewExpression'
+    else
+      'CallExpression'
+
+  astProperties: ->
+    return
+      callee: @variable.ast()
+      arguments: arg.ast() for arg in @args
+      optional: !!@soak
```

Comment:
`optional` is similar to what we saw with `MemberExpression` (this is how Babel was representing the proposed JS version of soak operators, though they now seem to have switched to distinguishing via `type` (here would be `OptionalCallExpression`) as of Babel 7)

Reply:
Probably, but I'd like to see if I can find an explanation as to why they switched

Reply:
If you're curious, https://github.com/babel/babel/issues/7256 discusses the switch (to `OptionalMemberExpression`/`OptionalCallExpression`). Haven't finished reading through it yet, but one thing that has implications if we want to mimic it is that `OptionalMemberExpression` "cascades" (so eg in `a?.b.c`, both the inner (ie `a?.b`) and outer (`a?.b.c`) are `OptionalMemberExpression`s - the inner has `optional: true` corresponding to it actually having a `?`, while the outer has `optional: false`)

Reply:
@GeoffreyBooth yes I think so. But I'd like to tackle that as a separate PR as there's some complexity there. And in the meantime I'll keep putting up individual node class PRs

Reply:
@GeoffreyBooth sure added a TODO comment on a test that should end up using `OptionalMemberExpression` (unless you meant commit a failing test?)

Reply:
I was thinking we'd add a commented out failing test, but either way works.